lib/source/Instruction.cpp

```diff
@@ -1575,18 +1575,26 @@ spvgentwo::Instruction* spvgentwo::Instruction::opConvertPtrToU(Instruction* _pP
 	return error();
 }
 
-spvgentwo::Instruction* spvgentwo::Instruction::opConvertUToPtr(Instruction* _pUInt,
-	const Type& _type)
+spvgentwo::Instruction* spvgentwo::Instruction::opConvertUToPtr(Instruction* _pResultType, Instruction* _pUInt)
 {
-	if (_type.isPointer() && _type.getStorageClass() == spv::StorageClass::PhysicalStorageBuffer)
+	const Type* resultType = _pResultType->getType();
```

Comment:
please check if type is not a nullptr before de-referrencing it. see https://github.com/rAzoR8/SpvGenTwo/blob/246b09a93cff37c14afcd7af5653a1b99d4069b6/lib/source/Instruction.cpp#L655 how I usually do it. The user might accidentally pass a instruction without a result type so getType() might return nullptr.